spec/integration_spec.rb

```diff
@@ -80,6 +87,44 @@
     end
   end
 
+  context "using plain tcp connection" do
+    it "should successfully connect to the server if ssl explicitely disabled" do
+      expect {
+        Lumberjack::Client.new(:port => tcp_port, :host => host, :addresses => host, :ssl => false)
+      }.not_to raise_error
+    end
+
+    it "should fail to connect to the server if ssl not explicitely disabled" do
+      expect {
+        Lumberjack::Client.new(:port => tcp_port, :host => host, :addresses => host)
+      }.to raise_error(RuntimeError, /Must set a ssl certificate/)
+    end
+
+    context "When transmitting a payload" do
+      let(:random_number_of_events) { Flores::Random.integer(2..10) }
+      let(:payload) { { "line" => "foobar" } }
+      let(:client) do
+        Lumberjack::Client.new(:port => tcp_port, :host => host, :addresses => host, :ssl => false)
+      end
+      let(:batch_size) { Flores::Random.integer(1..1024) }
+      let(:batch_payload) do
+        batch = []
+        batch_size.times do |n|
+          batch <<  { "line" => "foobar #{n}" }
+        end
+        batch
+      end
+
+      context "when sequence start at 0" do
+        let(:sequence_start) { 0 }
+
+        include_examples "send payload"
+      end
+
+    end
+  end
+
+
```

Comment:
Also I think it would be a great to create a **shared_example**  from the context "When transmitting a payload" to run the same tests over a SSL and a non SSL connection, only the `client` let would change in theses scenario.